### PR TITLE
LC30CHALLENGE: Second problem

### PR DESCRIPTION
**Pull request by DmitryBokovikov**

### CODE REVIEW

File: src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include "stdafx.h"
+#include "problem2.h"
+
+namespace problems_ns
+{
+    bool isHappy(int n)
+    {
+	constexpr int MAX_SUM_OF_DIGITS_SQUARE = 810;
```
Comment (fbokovikov) [inline src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp:8]: Could you elaborate why you use `810` as a restriction?

File: src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	std::vector<int> dp(MAX_SUM_OF_DIGITS_SQUARE + 1, -1);
+	dp[1] = 1;
+
+	auto get_sum_of_digits_square = [](int k)
+	{
+	    int sum = 0;
+	    while (k)
```
Comment (fbokovikov) [inline src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp:15]: Is it false only when k is 0?

Comment (DmitryBokovikov) [reply src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp:15]: exactly

File: src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp (hunk 1 above)
Comment (DmitryBokovikov) [reply src/main/cplusplus/leetcode_challenge/leetcode_challenge/problem2.cpp:8]: max int = 2,147,483,647 <= 9 999 999 999 sum of digits square for 9 999 999 999 = 81 * 10 = 810

**Pull request merged**